src/main/java/cn/edu/tsinghua/iotdb/benchmark/tsdb/iotdb/IoTDBClusterSession.java

```diff
@@ -117,34 +140,49 @@ public Status insertOneBatchByRecords(Batch batch) {
       typesList.add(constructDataTypes(record.getRecordDataValue().size()));
     }
 
+    AtomicBoolean sessionBroken = new AtomicBoolean(false);
+    calculateCurrentSession();
     future = service.submit(() -> {
       try {
         sessions[currSession]
             .insertRecords(deviceIds, times, measurementsList, typesList, valuesList);
-      } catch (IoTDBConnectionException | StatementExecutionException e) {
+      } catch (IoTDBConnectionException e) {
```

Comment:
should rethrow exception to make task end faster. This will help future.get throw ExecuteException not TimeOutException.